pkg/sql/txn_state.go

```diff
@@ -193,10 +193,6 @@ func (ts *txnState) resetForNewSQLTxn(
 	// Put the new span in the context.
 	txnCtx := opentracing.ContextWithSpan(connCtx, sp)
 
-	if !tracing.IsRecordable(sp) {
-		log.Fatalf(connCtx, "non-recordable transaction span of type: %T", sp)
-	}
```

Comment:
It is possible that the span created might be a noop (non-recordable) now because of random sampling. We do not need to check this anymore.